src/System.IO.Pipelines.File/FileReader.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Pipelines.File
+{
+    public class FileReader : PipelineReader
+    {
+        public FileReader(MemoryPool pool) : base(pool)
+        {
+        }
+
+        public FileReader(PipelineReaderWriter input) : base(input)
+        {
+        }
+
+        // Win32 file impl
+        // TODO: Other platforms
+        public unsafe void OpenReadFile(string path)
```

Comment:
Do you still need this now that FileStream has an optimized CopyToAsync implementation?  What further changes would be necessary there to obviate the need for this here?